onnxruntime/core/providers/acl/nn/conv.cc

```diff
@@ -0,0 +1,283 @@
+  bool acl_activ_enabled = false;
+
+  if (conv_attrs_.activation == "Relu") {
+    acl_activ_func = arm_compute::ActivationLayerInfo::ActivationFunction::RELU;
+    acl_activ_enabled = true;
+  } else if (conv_attrs_.activation == "LeakyRelu") {
+    acl_activ_func = arm_compute::ActivationLayerInfo::ActivationFunction::LEAKY_RELU;
+    acl_activ_enabled = true;
+  } else if (conv_attrs_.activation == "Tanh") {
+    acl_activ_func = arm_compute::ActivationLayerInfo::ActivationFunction::TANH;
+    acl_activ_enabled = true;
+  } else if (conv_attrs_.activation == "Sigmoid") {
+    acl_activ_func = arm_compute::ActivationLayerInfo::ActivationFunction::LOGISTIC;
+    acl_activ_enabled = true;
+  } else if (!conv_attrs_.activation.empty()){
+    ORT_NOT_IMPLEMENTED("Not implemented fused activation: ", conv_attrs_.activation);
+  }
+
+  ACLNEConv* pConv;
+  ConvLayersIterator it = Conv::convLayers.find((OpKernel*)this);
+  if (it == Conv::convLayers.end()) {
+    auto mm_layer = createMemoryManager();
+
+    ACLNEConv tconv;
+    tconv.mm_layer = std::move(mm_layer);
+
+    tconv.in = std::make_shared<arm_compute::Tensor>();
+    tconv.k = std::make_shared<arm_compute::Tensor>();
+    if (B != nullptr)
+      tconv.b = std::make_shared<arm_compute::Tensor>();
+    tconv.out = std::make_shared<arm_compute::Tensor>();
+
+    tconv.in->allocator()->init(arm_compute::TensorInfo(ACLTensorShape(X->Shape(), PREF_DIM), arm_compute::Format::F32));
+    tconv.k->allocator()->init(arm_compute::TensorInfo(ACLTensorShape(W->Shape()), arm_compute::Format::F32));
+    if (B != nullptr) {
+      tconv.b->allocator()->init(arm_compute::TensorInfo(ACLTensorShape(B->Shape()), arm_compute::Format::F32));
+    }
+    tconv.out->allocator()->init(arm_compute::TensorInfo(ACLTensorShape(Y->Shape(), PREF_DIM), arm_compute::Format::F32));
+
+    const arm_compute::DataLayout data_layout = tconv.in->info()->data_layout();
+    const int idx_channel = arm_compute::get_data_layout_dimension_index(data_layout, arm_compute::DataLayoutDimension::CHANNEL);
+    bool isDeptwise = (1 == tconv.k->info()->tensor_shape()[idx_channel]);
+
+    std::vector<int64_t> aclStrides(2);
+    aclStrides[0] = (strides.size() == 2) ? strides[1] : 1;
+    aclStrides[1] = strides[0];
+
+    std::vector<int64_t> aclPads(4);
+    if (pads.size() == 2) {
+      if (strides.size() == 1) {
+	aclPads[0] = 0;
+	aclPads[1] = 0;
+	aclPads[2] = pads[1];
+	aclPads[3] = pads[0];
+      } else {
+	aclPads[0] = pads[1];
+	aclPads[1] = pads[0];
+	aclPads[2] = pads[1];
+	aclPads[3] = pads[0];
+      }
+    } else {
+	aclPads[0] = pads[1];
+	aclPads[1] = pads[0];
+	aclPads[2] = pads[3];
+	aclPads[3] = pads[2];
+    }
+
+    arm_compute::PadStrideInfo aclPadStride = arm_compute::PadStrideInfo(aclStrides[0], aclStrides[1],
+	aclPads[0], aclPads[1], aclPads[2], aclPads[3], arm_compute::DimensionRoundingType::FLOOR);
+
+    if (isDeptwise) {
+#ifdef DEPTHWISE_CPU
```

Comment:
when would we need to use DEPTHWISE_CPU?

Reply:
DEPTHWISE implementation from ACL EP if functional but for the moment it is below CPU EP performance. I propose to keep it disabled until we manage to improve its performance.